File: adb-kotlin/src/main/java/co/herod/adbwrapper/testing/AndroidTestHelper.java
package co.herod.adbwrapper.testing;

import org.jetbrains.annotations.Nullable;

import java.util.List;
import java.util.concurrent.TimeUnit;
import java.util.function.Predicate;

import co.herod.adbwrapper.model.AdbUiNode;
import kotlin.jvm.functions.Function1;

/**
 * Created by matthewherod on 05/09/2017.
 */

public interface AndroidTestHelper {

    void assertActivityName(String activityName);

    void assertPower(int minPower);

    void assertValidApk(@Nullable String apkPath);

    void backButton();

    void closeLeftDrawer();

    void connectDevice();

    void dismissDialog();

    void dragDown(Function1<Integer, Integer> widthFunction);

    void dragUp(Function1<Integer, Integer> widthFunction);

    void failOnText(String text);

    void failOnText(String text, int timeout, TimeUnit timeUnit);

    List<String> getInstalledPackages();

    String getPackageVersionName(String packageName);

    void installApk(String apkPath);

    boolean installedPackageIsVersion(String packageName, String versionName);

    void launchApp(String packageName);

    void launchUrl(String url);

    void takeScreenshot();

    void touchText(String text);

    void assertScreenOn();

    void assertScreenOff();

    void turnScreenOn();

    void turnScreenOff();

    void uninstallPackage(String packageName);

    void updateApk(String apkPath);

    void waitForText(String text);

    void waitForText(String text, int timeout, TimeUnit timeUnit);

    void waitForUiNode(Predicate<AdbUiNode> adbUiNodePredicate);

    void waitSeconds(int waitSeconds);
}

File: adb-kotlin/src/main/java/co/herod/adbwrapper/testing/AndroidTestHelper.kt
package co.herod.adbwrapper.testing

import co.herod.adbwrapper.model.AdbUiNode
import java.util.concurrent.TimeUnit
import java.util.function.Predicate

/**
 * Created by matthewherod on 05/09/2017.
 */

@Suppress("RedundantVisibilityModifier")
public interface AndroidTestHelper {

    fun assertActivityName(activityName: String)

    fun assertNotActivityName(activityName: String)

    fun assertPower(minPower: Int)

    fun assertValidApk(apkPath: String?)

    fun backButton()

    fun closeLeftDrawer()

    fun connectDevice()

    fun dismissDialog()

    fun dismissKeyboard()

    fun dragDown(widthFunction: Function1<Int, Int>, edgeOffset: Double = 0.2)

    fun dragUp(widthFunction: Function1<Int, Int>, edgeOffset: Double = 0.2)

    fun dragLeft(heightFunction: Function1<Int, Int>, edgeOffset: Double = 0.2)

    fun dragRight(heightFunction: Function1<Int, Int>, edgeOffset: Double = 0.2)

    fun failOnText(text: String)

    fun failOnText(text: String, timeout: Int = 30, timeUnit: TimeUnit = TimeUnit.SECONDS)

    fun getInstalledPackages(): List<String>

    fun getPackageVersionName(packageName: String): String?

    fun installApk(apkPath: String)

    fun installedPackageIsVersion(packageName: String, versionName: String): Boolean

    fun forceStopApp(packageName: String)

    fun launchApp(packageName: String)

    fun launchUrl(url: String)

    fun launchUrl(url: String, packageName: String)

    fun takeScreenshot()

    fun touchText(text: String)

    fun typeText(text: String)

    fun assertScreenOn()

    fun assertScreenOff()

    fun turnScreenOn()

    fun turnScreenOff()

    fun uninstallPackage(packageName: String)

    fun updateApk(apkPath: String)

    fun waitForTextToDisappear(text: String)

    fun waitForTextToFailToDisappear(text: String)

    fun waitForText(text: String, timeout: Int = 30, timeUnit: TimeUnit = TimeUnit.SECONDS)

    fun waitForUiNode(adbUiNodePredicate: Predicate<AdbUiNode>)
    fun waitSeconds(waitSeconds: Int = 3)

    fun touchUiNode(adbUiNodePredicate: Predicate<AdbUiNode>)
}
